### PR TITLE
[border-router] use `Ip6::Nd` namespace for IPv6 Neighbor Discovery types

### PR DESCRIPTION
**Pull request by abtink**
This commit uses `Ip6::Nd` namespace for IPv6 Neighbor Discovery
definitions (used for border router). It also renames and moves
the source files to `net/nd6.*`.

### CODE REVIEW

File: src/core/border_router/nd6_types.hpp [hunk 1]
```diff
@@ -58,7 +58,7 @@ using ot::Encoding::BigEndian::HostSwap32;
 
 namespace ot {
 namespace BorderRouter {
-namespace RouterAdv {
+namespace Nd6 {
```
Comment (jwhui) [inline src/core/border_router/nd6_types.hpp:61]: Should these definitions be moved to https://github.com/openthread/openthread/blob/main/src/core/net/icmp6.hpp ?

Or at least under a `Ip6::Icmp6` namespace?

Comment (abtink) [reply src/core/border_router/nd6_types.hpp]: I moved it under `Ip6::Nd` namespace (to be similar to `Ip6::Udp`, `Ip6::Tcp` and `Ip6::Icmp`, etc.).
Moved the src files also under `net` folder.

Comment (jwhui) [reply src/core/border_router/nd6_types.hpp]: Do we want to move `Ip6::Nd` under `Ip6::Icmp`? Given that ND is built on top of ICMPv6?

File: src/core/net/nd6.hpp [hunk 2]
```diff
@@ -28,19 +28,17 @@
 
 /**
  * @file
- *   This file includes definitions for IPv6 Router Advertisement.
+ *   This file includes definitions for for IPv6 Neighbor Discovery (ND).
```
Comment (jwhui) [inline src/core/net/nd6.hpp:31]:
```suggestion
 *   This file includes definitions for IPv6 Neighbor Discovery (ND).
```

File: src/core/net/nd6.hpp [hunk 3]
```diff
@@ -256,15 +254,15 @@ class PrefixInfoOption : public Option, private Clearable<PrefixInfoOption>
...
      *
      */
-    void SetPrefix(const Ip6::Prefix &aPrefix);
+    void SetPrefix(const Prefix &aPrefix);
 
     /**
      * This method gets the prefix in this option.
      *
-     * @param[out] aPrefix   Reference to an `Ip6::Prefix` to return the prefix.
+     * @param[out] aPrefix   Reference to an `Prefix` to return the prefix.
```
Comment (jwhui) [inline src/core/net/nd6.hpp:262]:
```suggestion
     * @param[out] aPrefix   Reference to a `Prefix` to return the prefix.
```

File: src/core/net/nd6.hpp [hunk 4]
```diff
@@ -370,15 +368,15 @@ class RouteInfoOption : public Option, private Clearable<RouteInfoOption>
...
```
Comment (jwhui) [inline src/core/net/nd6.hpp:376]:
```suggestion
     * @param[out] aPrefix   Reference to a `Prefix` to return the prefix.
```

File: src/core/border_router/nd6_types.hpp (hunk 1 above)
Comment (jwhui) [reply src/core/border_router/nd6_types.hpp]: Yeah, I'm fine with how it is for now. If we expand ND further, we may want to consider aligning more with the protocol hierarchy.

**Review by jwhui — APPROVED** [review]

**Pull request merged**